### ISSUE TITLE
fio load: output encoding

### ISSUE THREAD

**Issue by heyarne** [Toblerity/Fiona#499]
## Expected behavior and actual behavior.

I have a GeoJSON that I'm outputting to a shapefile in order to change it's projection. The command I use is `cat input.geojson | fio load --src_crs EPSG:4326 --dst_crs EPSG:31467 --driver Shapefile output.shp'. During this process I see the following error:

```
WARNING:Fiona:CPLE_AppDefined in b"One or several characters couldn\'t be converted correctly from UTF-8 to ISO-8859-1.\\nThis warning will not be emitted anymore.
```

I was expecting the Shapefile to keep the original encoding or to be able to specify it. There must also be a way to reproject a Fiona Collection in my running script but I couldn't find it.

## Steps to reproduce the problem.

I uploaded the file I experienced this with [here](https://gist.github.com/heyarne/13d226d9900e63018f888e560dc164cb).

## Operating system

For example: Mac OS X 10.12.3.

## Fiona version and provenance

```
~$ conda --version
conda 4.3.23
~$ fio --version
fio, version 1.7.9
~$ python --version
Python 3.6.2
```

**Comment by sgillies** [comment]
I think the thing to do is add an `--encoding` option to fio-load and then pass the string value of  the option into `fiona.open()` at https://github.com/Toblerity/Fiona/blob/master/fiona/fio/load.py#L101. A handful of lines of code and a single new test should do it. Work should be done in the maint-1.7 branch and I then merged to master.

**Comment by heyarne** [comment]
I'd really like to help with this and I've written the code already, however I've got some failing tests due to missing files or invalid paths (my test output can be found [in this gist](https://gist.github.com/heyarne/c37c6e52f496b6f408bcebd1fbca4518)). Can I get the missing files somehow? I based my branch off of `maint-1.7`, revision `60aff7a758f58f3555430af7cc9559fe2e60ea54`.

**Comment by sgillies** [comment]
@heyarne the 1.7 tests have depended on nose, which creates the json file here: https://github.com/Toblerity/Fiona/blob/maint-1.7/tests/__init__.py#L33. If you install and run nosetests, I expect you'll get the needed file. After that you should be able to switch back to pytest if you prefer.

**Comment by sgillies** [comment]
@heyarne I'm making some related fixes in maint-1.7 and planning a 1.7.11 release. Would it be possible for you to send a PR soon, with or without tests, so I can include your work?

**Comment by heyarne** [comment]
I can have a look at it again tomorrow (EST).
